src/commands/install/install.py

```diff
@@ -0,0 +1,48 @@
+from enum import Enum
+from os import path
+from typing import Optional
+from git import InvalidGitRepositoryError, NoSuchPathError, Repo
+from git.objects import Submodule
+from typing_extensions import TypedDict
+from .utils import map_url_to_package_name, verify_repo_url
+
+
+class InstallationErrorEnum(Enum):
```

Comment:
In python we would preferably extend exception class and catch it where necessary using this class

Reply:
The developer doesn't know what errors can be thrown by a function when that approach is used. Therefore, in Elixir and in expo projects (TypeScript, TBV) errors that don't require including stack trace are usually returned rather than thrown.

I'll change it to use exception classes because to keep things consistent. Is there a naming convention for marking functions as throwable @Arcticae?